packages/admin-panel/src/table/DataFetchingTable/Pagination.jsx

```diff
@@ -0,0 +1,176 @@
+  border: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  border-radius: 4px;
+  padding-block: 0.5rem;
+  padding-inline: 0.8rem 0.2rem;
+  margin-inline: 0.5rem;
+  font-size: 0.75rem;
+  .MuiInputBase-input {
+    text-align: center;
+    padding-block: 0;
+    height: auto;
+  }
+`;
+
+const Text = styled(Typography)`
+  font-size: 0.75rem;
+`;
+
+const RowsSelect = styled(Select)`
+  display: flex;
+  flex-direction: row;
+  margin-block-end: 0;
+  margin-inline-end: 1.2rem;
+  width: 12rem;
+  .MuiSelect-root {
+    padding-block: 0.5rem;
+    padding-inline: 0.8rem 0.2rem;
+  }
+`;
+
+const PageSelectComponent = ({ onChangePage, page, pageCount }) => {
+  const pageDisplay = page + 1;
+  return (
+    <ActionsWrapper>
+      <ManualPageInputContainer>
+        <Text id="page-label" htmlFor="page" component="label">
+          Page
+        </Text>
+        <ManualPageInput
+          type="number"
+          value={pageDisplay}
+          onChange={e => {
+            const newPage = e.target.value ? Number(e.target.value) - 1 : '';
+            onChangePage(newPage);
+          }}
+          inputProps={{ min: 1, max: pageCount }}
+          aria-describedby="page-count"
+          id="page"
+          disableUnderline
+        />
+        <Text id="page-count">of {pageCount}</Text>
+      </ManualPageInputContainer>
+      <Button
+        onClick={() => onChangePage(page - 1)}
+        disabled={page === 0}
+        aria-label="Previous page"
+      >
+        <KeyboardArrowLeft />
+      </Button>
+      <Button
+        onClick={() => onChangePage(page + 1)}
+        disabled={page === pageCount - 1}
+        aria-label="Next page"
+      >
+        <KeyboardArrowRight />
+      </Button>
+    </ActionsWrapper>
+  );
+};
+
+PageSelectComponent.propTypes = {
+  onChangePage: PropTypes.func.isRequired,
+  page: PropTypes.number.isRequired,
+  pageCount: PropTypes.number.isRequired,
+};
+
+const RowsSelectComponent = ({ pageSize, setPageSize }) => {
+  const pageSizes = [5, 10, 20, 25, 50, 100];
+  return (
+    <ActionsWrapper>
+      <RowsSelect
+        value={pageSize}
+        onChange={event => setPageSize(Number(event.target.value))}
+        options={pageSizes.map(size => ({ label: `Rows per page: ${size}`, value: size }))}
+        variant="standard"
+      />
+    </ActionsWrapper>
+  );
+};
+
+RowsSelectComponent.propTypes = {
+  pageSize: PropTypes.number.isRequired,
+  setPageSize: PropTypes.func.isRequired,
+};
+
+export const Pagination = ({ page, pageCount, gotoPage, pageSize, setPageSize, totalRecords }) => {
+  if (!totalRecords) return null;
+  const currentDisplayStart = page * pageSize + 1;
+  const currentDisplayEnd = Math.min((page + 1) * pageSize, totalRecords);
+
+  const handleChangePage = newPage => {
+    gotoPage(newPage);
+  };
+  return (
+    <Wrapper>
+      <ActionsWrapper>
+        <Text>
+          {currentDisplayStart} - {currentDisplayEnd} of {totalRecords} entries
```

Comment:
Impossible to tell in code, but this is an en dash 👇🤓 (may need a comment)

```suggestion
          {currentDisplayStart} – {currentDisplayEnd} of {totalRecords} entries
```